pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1433,13 +1433,14 @@ PANDAS_INLINE void uppercase(char *p) {
 int to_boolean(const char *item, uint8_t *val) {
     char *tmp;
     int i, status = 0;
-    int bufsize = sizeof(char) * (strlen(item) + 1);
+    size_t length0 = (strlen(item) + 1);
+    int bufsize = length0;
 
     static const char *tstrs[1] = {"TRUE"};
     static const char *fstrs[1] = {"FALSE"};
 
     tmp = malloc(bufsize);
-    strncpy(tmp, item, bufsize);
+    snprintf(tmp,  length0, "%s", item);
```

Comment:
For my own edification: is there any potential performance difference here?

Reply:
potential? sure. effect size? not sure.  I'm not about to benchmark snprintf vs. strcpy for this silliness.
afaict, those are our choices:

- strncpy causes compiler warning, because this is not how it's meant to be used
- strcpy is flagged by cpplint, and there's no per-line ignore mechanism afaict. can disable printf checking for the whole file using a CPPLINT.cfg file.
- snprintf involves some overhead. at least "parsing" "%s", some input validation and an additional function call to strcpy. call it a few us per call.